engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -595,11 +601,34 @@ namespace dmGameSystem
             dmGameObject::HInstance instance_b = component_b->m_Instance;
             dmhash_t instance_a_id = dmGameObject::GetIdentifier(instance_a);
             dmhash_t instance_b_id = dmGameObject::GetIdentifier(instance_b);
+            uint64_t group_hash_a = GetLSBGroupHash(world, group_a);
+            uint64_t group_hash_b = GetLSBGroupHash(world, group_b);
 
-            dmPhysicsDDF::CollisionResponse ddf;
+            if (world->m_LuaListener != 0)
+            {
+                dmPhysicsDDF::Collision a;
+                a.m_Group =     group_hash_a;
+                a.m_Id =        instance_a_id;
+                a.m_Position =  dmGameObject::GetWorldPosition(instance_a);
+
+                dmPhysicsDDF::Collision b;
+                b.m_Group =     group_hash_b;
+                b.m_Id =        instance_b_id;
+                b.m_Position =  dmGameObject::GetWorldPosition(instance_b);
+
+                dmPhysicsDDF::CollisionEvent ddf;
+                ddf.m_A = a;
+                ddf.m_B = b;
```

Comment:
While it's small copy, it might not be always valid to do such a thing.
In this case, the struct is a POD type, but if it had contained an array, it would not have worked exactly as intended.

Normally, I think this copy is too small to care about but I think in terms of setting a good code practice on how to setup these events for posting, I would suggest setting up the CollisionEvent directly*
```
dmPhysicsDDF::CollisionEvent ddf;
dmPhysicsDDF::Collision& a = ddf.m_A;
a.m_Groupd = group_a;
...
```
I think it makes it just a more clear that there is a complexity to the structs, and that a simple copy might not be enough.

Reply:
oh, thank you! I totally missed that!